### PR TITLE
Refactored integrations (+ Fixed #2575)

### LINKED ISSUE #2575: ItemsAdder incompatibility
## :round_pushpin: Description (REQUIRED)

Well, the problem is, Slimefun isn't compatible with ItemsAdder plugin... 
That plugin adds custom blocks - Those blocks are really just an ArmorStand spawners with custom item on head, so blocks have custom texture. 
Well, when I break one of those custom blocks with Pickaxe of Containment from Slimefun, it drops 2 broken ArmorStand spawners. 

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)

- Place custom block from ItemsAdder plugin I mentioned before
- Mine it with Pickaxe of Containment
- It drops 2 broken spawners

## :bulb: Expected behavior (REQUIRED)

Since it's custom block, it shouldn't drop spawners, I think there should be some kind of check for items with custom model data in spawners or something like that 

## :scroll: Server Log

No errors

## :open_file_folder: /error-reports/ Folder

No errors

## :compass: Environment (REQUIRED)

This Server uses the following setup of Slimefun:
Tuinity v1_16_R3
Minecraft version 1.16.4
CS-CoreLib vDEV - 91 (git e42a05d1)
Slimefun vDEV - 756 (git 354dd30e)
Metrics build: #17

Installed Addons: (24)
GlobalWarming vDEV - 4 (git bd8e730a)
DyedBackpacks vDEV - 11 (git b8eda0c7)
ExtraTools vDEV - 22 (git ea11f88b)
FoxyMachines vDEV - 7 (git 85598948)
SoundMuffler vDEV - 17 (git 3abf8993)
SlimefunOreChunks vDEV - 10 (git 91df3b8e)
Residence v4.9.2.1
EcoPower vDEV - 2 (git b0c0c875)
HotbarPets vDEV - 31 (git 5909202f)
ExtraGear vDEV - 12 (git 4570ea2c)
MoreTools vDEV - 8 (git db8d8814)
SlimefunLuckyBlocks vDEV - 29 (git db79cb7c)
ExoticGarden vDEV - 61 (git b5067266)
RestrictedCreative v2.4.4
InfinityExpansion vDEV - 50 (git 255cc995)
LiteXpansion vDEV - 98 (git f79eab2f)
SoulJars vDEV - 17 (git 9eedba0f)
MobCapturer vDEV - 11 (git 4edf38dd)
SlimyTreeTaps vDEV - 15 (git 205f8afc)
PrivateStorage vDEV - 15 (git a0fbc572)
TranscEndence vDEV - 21 (git f7bb1376)
ColoredEnderChests vDEV - 17 (git f9a39e70)
FluffyMachines vDEV - 47 (git 22bbdb21)
ChestTerminal vDEV - 38 (git c024be5c)

ItemsAdder version 2.1.20

I hope I did everything correctly while reporting this bug, if not, let me know and I'll edit it, thanks :)

### PR DESCRIPTION
**Pull request by TheBusyBiscuit**
## Description

The existing `ThirdPartyPluginService` was messy at best.
This PR refactors this and moves them from the `core/services/plugins` package into a dedicated `integrations` package which is more logical location if you are looking for these.

I also fixed a few issues and added ItemsAdder support (see https://github.com/LoneDev6/ItemsAdder-API).

## Changes

* Refactored `ThirdPartyPluginService` into the `IntegrationsManager` class
* Moved all integrations into an `integrations` package
* Fixed #2575
* Added ItemsAdder compatibility

## Related Issues

* Resolves #2575 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Review by Sfiguz7 — APPROVED** [review]
I like the change, class looks a lot cleaner than what was there beforehand. I assume you want to move EmeraldEnchants support here too? Also, is this class supposed to just hold the *necessary* stuff or is any plugin Dev welcome to PR to it to ease integration? Granted Slimefun being open source and with an ever expanding API should allow anyone to provide support without needing to in most cases.

**Comment by TheBusyBiscuit** [comment]
EmeraldEnchants support used to be in that class, but it was removed a few builds ago in case you forgot.

And the goal of this class is the same as before, hook into any third party plugin.
Its open source so if anyone wants to pr something they obviously can...
But it probably will not really be modified a lot tbh.

File: src/main/java/io/github/thebusybiscuit/slimefun4/integrations/package-info.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/**
+ * This package holds classes which are related to integrations between Slimefun and Third-Party plugins.
+ */
+package io.github.thebusybiscuit.slimefun4.integrations;
```
Comment (svr333) [inline src/main/java/io/github/thebusybiscuit/slimefun4/integrations/package-info.java:4]: image
:)

**Review by svr333 — APPROVED** [review]
Looks good to me, couldn't find anything where I had doubts

File: src/main/java/io/github/thebusybiscuit/slimefun4/integrations/package-info.java (hunk 1 above)
Comment (TheBusyBiscuit) [reply src/main/java/io/github/thebusybiscuit/slimefun4/integrations/package-info.java:4]: It's just a docs file, not really too important XD

**Review by svr333 — APPROVED** [review]

**Comment by TheBusyBiscuit** [comment]
@WalshyDev @Sfiguz7

**Review by Sfiguz7 — APPROVED** [review]

**Comment by WalshyDev** [comment]
I was awaiting this TODO to be finished, I can review though if you want
https://github.com/Slimefun/Slimefun4/pull/2609/files#diff-a05e195a0fe48f7d9ee269be6161441c1eb82dc0c4a51b406c86af0aeaf064b1R49

**Comment by TheBusyBiscuit** [comment]
> I was awaiting this TODO to be finished, I can review though if you want
> https://github.com/Slimefun/Slimefun4/pull/2609/files#diff-a05e195a0fe48f7d9ee269be6161441c1eb82dc0c4a51b406c86af0aeaf064b1R49

That's for the next RC, we cannot refactor this now as it would break EG.
It needs to be changed in EG and SF simultaneously, so this is a reminder for RC-19, not part of the scope of this PR.

**Review by WalshyDev — APPROVED** [review]
LGTM

**Pull request merged**